test/lib/sensu-plugins-http/aws_v4_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'aws-sdk'
+require_relative '../../spec_helper.rb'
+require_relative '../../../lib/sensu-plugins-http/aws-v4'
+
+describe 'AwsV4' do
+  before :all do
+    @fake_http = Struct.new(:use_ssl?, :address, :port)
+  end
+
+  before :each do
+    ENV['AWS_REGION'] = nil
+    ENV['AWS_ACCESS_KEY_ID'] = nil
+    ENV['AWS_SECRET_ACCESS_KEY'] = nil
+    ENV['AWS_SESSION_TOKEN'] = nil
+
+    @aws_v4 = Object.new
+    @aws_v4.extend(SensuPluginsHttp::AwsV4)
+
+    @time_now = Time.utc(2015, 07, 06, 16, 48, 57)
+    allow(Time).to receive(:now).and_return(@time_now)
+  end
+
+  describe '#apply_v4_signature' do
+    it 'should apply v4 signature headers' do
+      ENV['AWS_REGION'] = 'us-east-2'
+      ENV['AWS_ACCESS_KEY_ID'] = 'ASIAJNP12RN4CST4GFFA'
```

Comment:
@eheydrick what is our stance on this kinda stuff I personally think it would be better to setup a keypair in travis via ENV than have this hardcoded?